ApplicationCommands/src/Program.cs

```diff
@@ -0,0 +1,133 @@
+{
+    public class Program
+    {
+        public static void Main(string[] args) => MainAsync(args).ConfigureAwait(false).GetAwaiter().GetResult();
+
+        public static async Task MainAsync(string[] args)
+        {
+            // Logging! Let the user know that the bot started!
+            Console.WriteLine("Starting bot...");
+            
+            // CHALLENGE: Try making sure the token is provided! Hint: A Try/Catch block may be needed!
+            DiscordConfiguration discordConfiguration = new()
+            {
+                // The token is recieved from the command line arguments (bad practice in production!)
+                // Example: dotnet run <someBotTokenHere>
+                // CHALLENGE: Make it read from a file, optionally from a json file using System.Text.Json
+                // CHALLENGE #2: Try retriving the token from environment variables
+                Token = args[0]
+            };
+
+            DiscordShardedClient discordShardedClient = new(discordConfiguration);
+
+            Console.WriteLine("Connecting to Discord...");
+            await discordShardedClient.StartAsync();
+
+            // Use the default logger provided for easy reading
+            discordShardedClient.Logger.LogInformation($"Connection success! Logged in as {discordShardedClient.CurrentUser.Username}#{discordShardedClient.CurrentUser.Discriminator} ({discordShardedClient.CurrentUser.Id})");
+
+            // Register a Random class instance now for use later over in RollRandom.cs
+            ApplicationCommandsConfiguration appCommandsConfiguration = new()
+            {
+                Services = new ServiceCollection().AddSingleton<Random>().BuildServiceProvider()
+            };
+
+            // Let the user know that we're registering the commands.
+            discordShardedClient.Logger.LogInformation("Registering application commands...");
+
+            // If the guild ID is not provided, then register global commands.
+            ulong? guildId = null;
+            if (args.Length > 1)
+            {
+                guildId = ulong.Parse(args[1]);
+            }
+
+            Type appCommandModule = typeof(ApplicationCommandsModule);
+            foreach (DiscordClient discordClient in discordShardedClient.ShardClients.Values)
+            {
+                ApplicationCommandsExtension appCommandShardExtension = discordClient.UseApplicationCommands(appCommandsConfiguration);
+
+                // Register event handlers
+                appCommandShardExtension.SlashCommandExecuted += Slash_SlashCommandExecuted;
+                appCommandShardExtension.SlashCommandErrored += Slash_SlashCommandErrored;
+                appCommandShardExtension.ContextMenuExecuted += Context_ContextMenuCommandExecuted;
+                appCommandShardExtension.ContextMenuErrored += Context_ContextMenuCommandErrored;
+                
+                appCommandShardExtension.RegisterCommands<Ping>(guildId);
```

Comment:
We shouldn't need to register the commands manually anymore. We can use reflection to grab the classes which inherit from `ApplicationCommand` and register them through the `Type` overload `RegisterCommands` has. To add role/user permissions, simply add an if statement in your foreach and add your logic in that.